admin/spec/back-end/controllers/check-window.spec.js

```diff
@@ -71,8 +71,35 @@ describe('access arrangements controller:', () => {
       spyOn(checkWindowV2AddService, 'submit')
       await controller.submitCheckWindow(req, res, next)
       expect(checkWindowV2AddService.submit).toHaveBeenCalled()
+      expect(next).not.toHaveBeenCalled()
+      expect(req.flash).toHaveBeenCalled()
       expect(res.redirect).toHaveBeenCalled()
     })
+    it('calls next when checkWindowV2AddService submit throws an error', async () => {
+      const res = getRes()
+      const req = getReq(reqParams)
+      spyOn(res, 'redirect')
+      spyOn(res, 'render')
+      const error = new Error('error')
+      error.name = 'error'
+      const unsafeReject = p => {
+        p.catch(ignore => ignore)
+        return p
+      }
+      const rejection = unsafeReject(Promise.reject(error))
+      spyOn(checkWindowV2AddService, 'submit').and.returnValue(rejection)
+      try {
+        await controller.submitCheckWindow(req, res, next)
```

Comment:
Should there be a `fail()` right after the line in the try block that is expected to throw?